api/python/quilt3/backends/s3.py

```diff
@@ -0,0 +1,20 @@
+from quilt3.data_transfer import list_url, get_bytes
+
+from .base import PackageRegistryV1
+
+
+class S3PackageRegistryV1(PackageRegistryV1):
+    def list_packages(self):
+        prev_pkg = None
+        for path, _ in list_url(self.pointers_global_dir):
+            pkg = path.rpartition('/')[0]
```

Comment:
`rpartition` is a strange choice as it will fail if there are multiple slashes in the (sub) package name? or can that never happen?

Reply:
We allow only a single slash in the package name, so it should work unless some put file `.quilt/named_packages/foo/bar/baz/bla`, but I think we assume that registry is not broken.

Reply:
OK